### PR TITLE
doc: zbus: update documentation with changes for 3.5

### PR DESCRIPTION
**Pull request by rodrigopex**
Add documentation for the new way to allocate observers, the message subscribers, and the confirmed message sample. And 
 add the breaking changes to the migration guide.

Structural changes: 

* Replace Zbus with ZBus;
* Remove the message from the slogan. Zephyr Bus instead of Zephyr Message Bus;
* Change the images.

### CODE REVIEW

**Comment by rodrigopex** [comment]
@kartben, I would like to share the Figma file I used to draw the bus diagrams with people. The files are here: https://www.figma.com/community/file/1292866458780627559. What do you think is the best way to do that?

File: doc/services/zbus/index.rst
Comment (kartben) [inline doc/services/zbus/index.rst:3]: If you want to spell it ZBus from now on please also consider updating other places where it might be referred to as "zbus", for example in code samples. FWIW I liked the lower-case naming :)

File: doc/services/zbus/index.rst
Comment (kartben) [inline doc/services/zbus/index.rst:14]: documentation should also following coding guidelines and have lines wrap at 100 characters. 
The long lines are making the actuall differences hard to review :/

Comment (kartben) [inline doc/services/zbus/index.rst:14]: I would also consider making this several paragraphs, as it is a bit intimidating to have the introduction to ZBus/zbus be such a big block of text

image

**Review by kartben — COMMENTED** [review]
added a few comments. Anything you could do to make the diff easier to review would be appreciated!

**Comment by kartben** [comment]
> @kartben, I would like to share the Figma file I used to draw the bus diagrams with people. The files are here: https://www.figma.com/community/file/1292866458780627559. What do you think is the best way to do that?

I would add a comment at the top of the index.rst

```
..
   Note to documentation authors: the diagrams included in this documentation page were designed
   using the following Figma library: xxx. 
```

File: doc/services/zbus/index.rst
Comment (rodrigopex) [reply doc/services/zbus/index.rst:14]: Do you suggest any formatting tool like `clang-format` to do this? I am trying to use the [rstfmt](https://pypi.org/project/rstfmt/), but it is not working because of the sphinx commands. I suggest standardizing that like we do with the code. I will try out something here to improve the documentation format and organization.

Comment (rodrigopex) [reply doc/services/zbus/index.rst:14]: The old and good Neovim (Lazyvim, in my case), with its `gw` keybinding, immensely helped.

Comment (rodrigopex) [reply doc/services/zbus/index.rst:14]: The only issue here is that the new line-split file will also be hard to keep track of because it will change all the text.

Comment (kartben) [reply doc/services/zbus/index.rst:14]: Just make a separate commit for the formatting-only changes?

Comment (rodrigopex) [reply doc/services/zbus/index.rst:14]: Ok. I will do.

Comment (rodrigopex) [reply doc/services/zbus/index.rst:14]: It will not be possible. There are tons of rework to do. I tried to do that, but there were many conflicts and other issues. I am rewriting a significant portion of that. It may take more time to review, but it would be better to check all to see if it is still consistent. Sorry for the burden.

**Review by rodrigopex — COMMENTED** [review]
Migration guide entries added.

**Review by jhedberg — DISMISSED** [review]

File: doc/releases/migration-guide-3.5.rst
Comment (jukkar) [inline doc/releases/migration-guide-3.5.rst:239]: The sentence
```
The configuration changed.
```
looks quite lonely. Is there something missing from it or perhaps more explanation could be added to it?

Comment (rodrigopex) [reply doc/releases/migration-guide-3.5.rst:239]: For sure. I will adjust that.

Comment (rodrigopex) [reply doc/releases/migration-guide-3.5.rst]: Done!

**Review by rodrigopex — COMMENTED** [review]
Improve migration guide sentence regarding runtime observers.

**Comment by rodrigopex** [comment]
@jukkar and @jhedberg, could you please take a look at that again? There was some conflict, and I needed to submit that again after accidentally accepting a merge offered by GitHub.

**Review by jhedberg — DISMISSED** [review]

File: doc/releases/migration-guide-3.5.rst
Comment (kartben) [inline doc/releases/migration-guide-3.5.rst:278]: nit: it probably doesn't make sense to use :kconfig: to point to an option that doesn't exist anymore

```suggestion
  - Replace the integer ``CONFIG_ZBUS_RUNTIME_OBSERVERS_POOL_SIZE`` with the boolean
```

File: doc/services/zbus/index.rst
Comment (kartben) [inline doc/services/zbus/index.rst:331]:
```suggestion
   :kconfig:`CONFIG_ZBUS_MSG_SUBSCRIBER_NET_BUF_POOL_SIZE` and :kconfig:`CONFIG_HEAP_MEM_POOL_SIZE`.
```

File: doc/services/zbus/index.rst
Comment (kartben) [inline doc/services/zbus/index.rst:450]: I am confused about the use of "code" formatting for `validator function`, is this on purpose? I think you probably meant to use single backquotes?

File: doc/services/zbus/index.rst
Comment (kartben) [inline doc/services/zbus/index.rst:455]: ditto

File: doc/services/zbus/index.rst
Comment (kartben) [inline doc/services/zbus/index.rst:786]: btw it makes me realize that it would maybe make sense to provide a Sphinx directive that outputs the equivalent of what shows up in the API reference, as this could be useful to list samples anywhere.

image

**Review by kartben — CHANGES_REQUESTED** [review]
Great stuff @rodrigopex! I based my review mostly on the looks of the final rendered page rather than looking specifically at the diff -- spotted a few formatting style issues.

File: doc/services/zbus/index.rst
Comment (rodrigopex) [reply doc/services/zbus/index.rst:786]: @kartben, how could you change the title of the note? 

image

Should I remove the Samples sections?

**Review by rodrigopex — COMMENTED** [review]
@kartben, I changed most of your requests. The only thing I still need to change is the samples. I'm just waiting for your feedback to remove that or not. But the rest is ok.

File: doc/services/zbus/index.rst
Comment (kartben) [reply doc/services/zbus/index.rst]: Sorry I should have been clearer, and I think it's fine if you live it as-is for now. I was mostly talking to myself and thinking that I should add to the code-sample sphinx extension a way to easily output the list of samples related to a specific set of doxygengroups, as it might be useful in situations like yours. 

That being said, this section pretty much "paraphrases" the list of samples that also already show up automatically in the API reference, mere lines below, so if you think that list is enough and easy enough to not miss for folks, then maybe just drop your Samples section altogether (but again, also 100% fine with me if you'd like to keep it!)

**Review by kartben — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Pull request merged**
